sklearn/feature_extraction/tests/test_feature_hasher.py

```diff
@@ -107,3 +107,18 @@ def test_hasher_zeros():
     # Assert that no zeros are materialized in the output.
     X = FeatureHasher().transform([{'foo': 0}])
     assert_equal(X.data.shape, (0,))
+
+
+def test_hasher_non_negative():
+    X = [["foo", "bar", "baz"]]
+
+    Xt = FeatureHasher(non_negative=False,
+                      input_type='string').fit_transform(X)
+    assert_true(Xt.data.min() < 0 and Xt.data.max() > 0)
+    Xt = FeatureHasher(non_negative=True,
+                      input_type='string').fit_transform(X)
+    assert_true((Xt.data >= 0).all())  # zeros are acceptable
+    Xt = FeatureHasher(non_negative='total',
+                      input_type='string').fit_transform(X)
+    assert_true((Xt.data > 0).all())  # strictly positive counts
```

Comment:
You are right, thanks, changed the test to actually have a hash collision so this part is tested.